app/routes/objects.js

```diff
@@ -0,0 +1,39 @@
+import Joi from 'joi'
+import { StatusCodes } from 'http-status-codes'
+
+import { handleObjectRetrieval } from '../services/retrieve.js'
+import { FILE_NOT_FOUND } from '../constants/file-errors.js'
+
+const handleRetrievalError = (err, h) => {
+  switch (err?.cause) {
+    case FILE_NOT_FOUND:
+      return h.response().code(StatusCodes.NOT_FOUND)
+    default:
+      throw err
+  }
+}
+
+const objects = {
+  method: 'GET',
+  path: '/objects/{id}',
+  handler: async (request, h) => {
+    const { id } = request.params
+
+    const [fileObject, err] = await handleObjectRetrieval(id)
+
+    if (err) {
+      return handleRetrievalError(err, h)
+    }
+
+    return h.response(fileObject).code(StatusCodes.OK)
+  },
+  options: {
```

Comment:
I would rearrange this so the options is above the handler, its easier to read and the validation will happen first. Also follows how we have routes set up across fcp projects